dlt/sources/helpers/rest_client/paginators.py

```diff
@@ -76,7 +76,139 @@ def update_request(self, request: Request) -> None:
         return
 
 
-class OffsetPaginator(BasePaginator):
+class BaseNumericPaginator(BasePaginator):
+    """A base paginator class for paginators that use a numeric parameter
+    for pagination, such as page number or offset.
+
+    See `PageNumberPaginator` and `OffsetPaginator` for examples.
+    """
+
+    def __init__(
+        self,
+        param_name: str,
+        initial_value: int,
+        total_path: jsonpath.TJsonPath,
```

Comment:
@rudolfix thanks for suggestion. I think it could be very useful to control the pagination. 
I've implemented the maximum_value in the [f7f7860](https://github.com/dlt-hub/dlt/pull/1307/commits/f7f7860683140f3b5591102118b0eabfc4b4e36d).
I also really like the idea of making maximum_value a substitute for total number – this way we make this paginator very versatile. Will send the update soon.